### DISCUSSION TITLE
Question about import/export

### DISCUSSION THREAD

**Discussion by saitoh183** [arabcoders/watchstate#356] [General] [answered]
Hi,
So I don't think I'm sure how the export and import works. I have a plex and emby that I want to make sure is sync so I can use either platform. I set up the webhook on both machine. If I want to update my plex with my emby watched, do I use the export command on emby or import command on emby to send to plex? I noticed some shows not being sent sometimes that have been watched for more than 24h so I wanted to do a manual sync but wasn't sure which command to use.

**Comment by arabcoders** [comment, marked as answer]
Hi please follow the steps outlined in 

https://github.com/ArabCoders/watchstate/blob/master/FAQ.md#my-new-backend-overriding-my-old-backend-state--my-watch-state-is-not-correct

It should do what you want. If not please update this post.

**Reply by saitoh183** [reply]
Before i proceed with the export command, does this configuration look right? Embylocal is the correct watched state (old backend) and Plexlocal is the new backend. I corrected all the playstate in Emby. i really dont want to mess this up as i dont want to have to unmark stuff on emby again.
┌───────────┬────────────────────────────────────────────────────────────────────────────────┐
│ Backend   │ Filter: None                                                                   │
├───────────┼────────────────────────────────────────────────────────────────────────────────┤
│ plexlocal │ name: plexlocal                                                                │
│           │ type: plex                                                                     │
│           │ token: '*HIDDEN*'                                                              │
│           │ url: 'https://172-20-0-100.8e16590023224aaf90ef1cc050788dd0.plex.direct:32400' │
│           │ uuid: 4b08a11a64e480c29823d9c6cf317f34ffbfc425                                 │
│           │ user: 1069973                                                                  │
│           │ import:                                                                        │
│           │   enabled: false                                                               │
│           │   lastSync: 1699495261                                                         │
│           │ export:                                                                        │
│           │   enabled: true                                                                │
│           │   lastSync: 1699493503                                                         │
│           │ webhook:                                                                       │
│           │   match:                                                                       │
│           │     user: true                                                                 │
│           │     uuid: true                                                                 │
│           │   token: '*HIDDEN*'                                                            │
│           │ options:                                                                       │
│           │   IMPORT_METADATA_ONLY: false                                                  │
├───────────┼────────────────────────────────────────────────────────────────────────────────┤
│ embylocal │ name: embylocal                                                                │
│           │ type: emby                                                                     │
│           │ url: 'http://emby:8096'                                                        │
│           │ token: '*HIDDEN*'                                                              │
│           │ uuid: ec3d761b216040f5aa86080a10b29623                                         │
│           │ user: 6bb7ada937cd4343ab1881db00a1825f                                         │
│           │ import:                                                                        │
│           │   enabled: true                                                                │
│           │   lastSync: 1699495262                                                         │
│           │ export:                                                                        │
│           │   enabled: false                                                               │
│           │   lastSync: 1699493503                                                         │
│           │ webhook:                                                                       │
│           │   match:                                                                       │
│           │     user: false                                                                │
│           │     uuid: false                                                                │
│           │   token: '*HIDDEN*'                                                            │
└───────────┴────────────────────────────────────────────────────────────────────────────────┘

**Reply by arabcoders** [reply]
Looks correct to me.

**Reply by saitoh183** [reply]
Once it is done do i have to have both backends with import and export set to enabled? To keep everything in sync

**Comment by saitoh183** [comment]
Well it did replace my watched status in plex but i found some mistakes that i corrected in emby and ran the command 
`docker exec -ti watchstate console state:export -vvif -s plexlocal` again but it doesnt seem to update the shows i modified in Emby
last line at the end of the log: 
`[2023-11-09T02:01:45-05:00] INFO: Parsing [plexlocal] library [TV Shows] [62/62] response is complete.`

`[2023-11-09T02:01:45-05:00] NOTICE: SYSTEM: Sent [99] play state comparison requests.`

`[2023-11-09T02:01:45-05:00] NOTICE: Export mode ends.`

`[2023-11-09T02:01:45-05:00] NOTICE: SYSTEM: No play state changes detected.`

example of a show i changed but is still not changing in plex
Emby:
![image](https://github.com/ArabCoders/watchstate/assets/1534197/1a6b1565-0148-4181-b79d-1eed7aa945c3)

Plex
![image](https://github.com/ArabCoders/watchstate/assets/1534197/7088e0a0-df50-463d-9ff6-1c05747ff236)

**Reply by arabcoders** [reply]
After changing the item state you need to import it and then export if you didn't enable the task scheduler

**Reply by saitoh183** [reply]
ok seems most things are now synced. Do i set both to backends to import and export enabled now to keep them in sync?

**Reply by arabcoders** [reply]
Yeah, Just make sure to enable the import & export tasks. might as well enable the backup task as well it's really helpful in case something goes wrong

**Reply by saitoh183** [reply]
Thanks for the help. I just find the terminology a bit confusing with the export and import. When i hear Export, i think of exporting stuff from source (ex: Export from Embylocal) but you use it the opposite way and mean send already exported data to new backend. I think making a diagram might help people figure because im sure im not the only one that thinks that...or maybe i am :)...lol

**Reply by arabcoders** [reply]
Lol, probably yeah, I honestly need to move away from the cli interface to a proper GUI so poeple are less confused, i just don't have the energy to do it. 

In my mind, `import`  i.e. `importing` is getting something from somewhere, `export` as in exporting is to send something somewhere, it's helps to understand that the tool itself is a database when you `import` things you import things into the database, and then `export` send the data in proper format back to the selected backends.
